cases/forms/case.py

```diff
@@ -138,10 +150,35 @@ def clean_type(self):
         type = ActionType.objects.get(id=type)
         return type
 
+    def clean(self):
+        if self._is_time_in_future():
+            raise ValidationError("The time of the action can't be in the future.")
+        return self.cleaned_data
+
     def save(self, case):
+        self.instance.time = self._get_time_from_cleaned_data()
         self.instance.case = case
         super().save()
 
+    def _is_time_in_future(self):
+        time_now = now()
+
+        if "date" in self.cleaned_data and self.cleaned_data["date"] > time_now.date():
+            return True
+
+        if set(["date", "action_time"]).issubset(self.cleaned_data.keys()):
+            return self._get_time_from_cleaned_data() > time_now
+
+        return False
+
+    def _get_time_from_cleaned_data(self):
+        return datetime.combine(
+            self.cleaned_data["date"],
+            self.cleaned_data["action_time"],
+            # Hack to get the local timezone.
```

Comment:
Replace with `timezone.get_current_timezone().localize(time)` after the combine.